pysteps/tests/test_io_knmi_hdf5.py

```diff
@@ -10,54 +11,40 @@
 pytest.importorskip("h5py")
 
 
-def test_io_import_knmi_hdf5_shape():
-    """Test the importer KNMI HDF5."""
-
-    root_path = pysteps.rcparams.data_sources["knmi"]["root_path"]
+root_path = pysteps.rcparams.data_sources["knmi"]["root_path"]
+filename = os.path.join(root_path, "2010/08", "RAD_NL25_RAP_5min_201008260000.h5")
+data_array = pysteps.io.import_knmi_hdf5(filename)
 
-    filename = os.path.join(root_path, "2010/08", "RAD_NL25_RAP_5min_201008260000.h5")
 
-    precip, __, __ = pysteps.io.import_knmi_hdf5(filename)
-
-    assert precip.shape == (765, 700)
+def test_io_import_knmi_hdf5_shape():
+    """Test the importer KNMI HDF5."""
+    assert isinstance(data_array, xr.DataArray)
+    assert data_array.shape == (765, 700)
 
 
 # test_metadata: list of (variable,expected, tolerance) tuples
 
-expected_proj1 = (
+expected_proj = (
     "+proj=stere +lat_0=90 +lon_0=0.0 "
     "+lat_ts=60.0 +a=6378.137 +b=6356.752 +x_0=0 "
     "+y_0=0"
 )
 
-test_metadata = [
-    ("projection", expected_proj1, None),
-    ("x1", 0.0, 1e-10),
-    ("y1", -4415.038179210632, 1e-10),
-    ("x2", 699.9842646331593, 1e-10),
-    ("y2", -3649.950360247753, 1e-10),
-    ("xpixelsize", 1.0, 1e-10),
-    ("xpixelsize", 1.0, 1e-10),
-    ("cartesian_unit", "km", None),
+# list of (variable,expected,tolerance) tuples
+test_attrs = [
+    ("projection", expected_proj, None),
+    ("institution", "KNMI - Royal Netherlands Meteorological Institute", None),
     ("accutime", 5.0, 1e-10),
-    ("yorigin", "upper", None),
     ("unit", "mm", None),
-    ("institution", "KNMI - Royal Netherlands Meteorological Institute", None),
     ("transform", None, None),
     ("zerovalue", 0.0, 1e-10),
     ("threshold", 0.01, 1e-10),
-    ("zr_a", 200.0, None),
-    ("zr_b", 1.6, None),
+    ("zr_a", 200.0, 0.1),
+    ("zr_b", 1.6, 0.1),
 ]
 
 
-@pytest.mark.parametrize("variable,expected,tolerance", test_metadata)
+@pytest.mark.parametrize("variable,expected,tolerance", test_attrs)
 def test_io_import_knmi_hdf5_metadata(variable, expected, tolerance):
     """Test the importer KNMI HDF5."""
-    root_path = pysteps.rcparams.data_sources["knmi"]["root_path"]
-
-    filename = os.path.join(root_path, "2010/08", "RAD_NL25_RAP_5min_201008260000.h5")
-
-    __, __, metadata = pysteps.io.import_knmi_hdf5(filename)
-
-    smart_assert(metadata[variable], expected, tolerance)
+    smart_assert(data_array.attrs[variable], expected, tolerance)
```

Comment:
Now that x, y, pixelsize, etc. is part of the DataArray and not an attribute or metadata anymore (and thus not tested as such), should the KNMI importer also have a 'geodata' that is tested here, similar to the other importers?

Reply:
See #218: the idea for V2 is to remove the national importers from the core library (pysteps) and refactor them as optional plugins. In this sense, the KNMI will become an external package (e.g. "pysteps-knmi-importer") that will be maintained (and tested) externally (by you I suppose ;-) )